File: Palindromic Substrings.py
class Solution:
    def countSubstrings(self ,arr):
        res=""
        reslen=0
        count=0
        for i in range(len(arr)):
            l=i
            r=i
            while l>=0 and r<len(arr) and arr[l]==arr[r]:
                if r-l+1>reslen:
                    res=arr[l:r+1]
                    reslen=r-l+1
                count+=1
                l-=1
                r+=1
            l=i
            r=i+1     
            while l>=0 and r<len(arr) and arr[l]==arr[r]:
                if r-l+1>reslen:
                    res=arr[l:r+1]
                    reslen=r-l+1
                count+=1
                l-=1
                r+=1
        return count
        

File: Java/Palindromic Substrings.java
class Solution {
    public int countSubstrings(String s) {
        int count=0;
        int n=s.length();
        int L;
        for(int i=0 ; i < s.length();i++){
            L=i;

            count+=expand(L,L,n,s);
            count+=expand(L,L+1,n,s);
        }

        return count;
        
    }

    public int expand(int L , int R , int n, String s){
        int count1=0;
        while (L>=0 && R<n){
            if(s.charAt(L)!=s.charAt(R)){
                break;
            }
            count1++;
            L--;
            R++;
        }
        return count1;


    }
}
